pytorch_lightning/trainer/_progress.py

```diff
@@ -0,0 +1,113 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+
+
+@dataclass
+class LoopProgress:
+    """Basic dataclass to track loop progress during execution.
+
+    The general structure is to count totals across epochs as well as intra-epoch counts.
+    These counters are local to a trainer rank.
+    By default, they are not by default globally synced across all ranks.
+
+    Args:
+        total_epochs_finished: Number of passes through the dataset.
+        total_batches_read: Number of batches loaded through the dataloader.
+        total_batches_started: Number of batches started processing.
+            This is intended to be incremented after `on_train_batch_start` completes.
```

Comment:
```
on_train_batch_start
increment_total_batches_started
training_step
```
you mean this right? 

The call_hook forces the trainer's hook to run first. This means callbacks run before the model hook which runs before this increment. I think this can lead to confusion, so the ordering is something we have to be careful about